.github/workflows/msbuild.yml

```diff
@@ -0,0 +1,82 @@
+name: MSBuild
+
+on:
+  push:
+    branches: ["master"]
+  pull_request:
+    branches: ["master"]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        configuration: [Debug, Release]
+
+    runs-on: windows-latest
+
+    env:
+      BUILD_CONFIGURATION: ${{matrix.configuration}}
+      # Path to the solution file relative to the root of the project.
+      SOLUTION_FILE_PATH: ./mhw-cs-plugin-loader.sln
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Add msbuild to PATH
+        uses: microsoft/setup-msbuild@v1.1
+        with:
+          vs-version: "[,17)"
+          msbuild-architecture: x64
+
+      - name: Install vcpkg
+        uses: lukka/run-vcpkg@v10.2
+        with:
+          vcpkgDirectory: "${{runner.workspace}}/b/vcpkg"
+          vcpkgGitCommitId: "c8696863d371ab7f46e213d8f5ca923c4aef2a00" # github.com/microsoft/vcpkg - 2023-12-12 release.
+          vcpkgJsonGlob: "**/vcpkg.json"
+          runVcpkgInstall: true
+
+      - name: Integrate vcpkg
+        working-directory: ${{env.VCPKG_ROOT}}
+        run: |
+          vcpkg integrate install
+
+      - name: Generate cimgui project files
+        working-directory: ${{env.GITHUB_WORKSPACE}}
+        run: |
+          cd ./dependencies/cimgui/
+          cmake -B . -G 'Visual Studio 17 2022'
+
+      - name: Dotnet Restore
+        working-directory: ${{env.GITHUB_WORKSPACE}}
+        run: |
+          dotnet restore ${{env.SOLUTION_FILE_PATH}}
+
+      - name: Build Utilities\ChunkBuilder (Release mode)
+        working-directory: ${{env.GITHUB_WORKSPACE}}
+        # We _ALWAYS_ build ChunkBuilder in Release mode, as certain build scripts expect a release build path on disk,
```

Comment:
We should probably change that so it just uses the one for the selected configuration